visualization/app/codeCharta/ui/fileChooser/fileChooser.component.ts

```diff
@@ -39,9 +39,7 @@ export class FileChooserController {
 
 				reader.onload = event => {
 					if (isCompressed) {
-						const zlib = require("zlib")
-
-						content = zlib.unzipSync(Buffer.from(event.target.result))
+						content = require("zlib").unzipSync(Buffer.from(event.target.result))
```

Comment:
I don't know, why this has to be an inlined-require block.
If I import zlib with es6-imports, the e2e tests are failing with an error, that require is not defined. The tests somehow passed in travis though.

Reply:
You found an actual bug. I guess it's somewhat my fault that I did not see that properly before. This is a Node.js module but the frontend has no `zlib` module. I was originally mistaken where this code would run.

Thus, we have to use some kind of module that is able to handle the data.

Reply:
Ehm, ok. I still don't understand. If the frontend has no zlib module, why can we require it though?
What is our context actually? Are we in some DOM-context or is angular actually node-application?

Reply:
I have no idea why it did not fail earlier. Do we actually reach this code in our test?

Reply:
OH. That's why it's working. No, we never reach it.
But we'd reach an import on top. That makes sense

Reply:
But still: Why is this working in production?

Reply:
I am not certain what you mean with it working in production?

I'd say you found another bug: we can't choose zipped files anymore :) Those should be available.

Reply:
Our angular tests run in the `jsdom` environment. [Puppeteer chooses it's environment ](https://github.com/smooth-code/jest-puppeteer#usage). Afaik the puppeteer environment extends the NodeEnvironment. I thought zlib would be available then.

Our [webpack config](https://github.com/MaibornWolff/codecharta/blob/main/visualization/conf/webpack.config.js) compils our code to two different targets. We can create a node environment for the standalone (this allows us to use the cool upcoming features for standalone only, e.g. opening files, when double clicking a building) and we can create a [browser-like environment](https://webpack.js.org/configuration/target/) using `web`.
I think you can open a zipped `cc.json` with both targets.

Reply:
No, the compressed detection checks for the file ending. `json` is not compressed.

We are indeed able to do more in the standalone version but that should not limit the web version. So maybe let's just fix that? :-)

Reply:
We load browserify at some point and browserify has a zlib implementation that it ships. This causes issues when loading files are not compressed gzipped:

```
Uncaught Error: incorrect header check
    at Zlib._handle.onerror (index.js:352)
    at Zlib._error (binding.js:283)
    at Zlib._checkError (binding.js:254)
    at Zlib._write (binding.js:116)
    at Zlib.writeSync (binding.js:77)
    at Unzip.Zlib._processChunk (index.js:511)
    at zlibBufferSync (index.js:236)
    at Object.exports.unzipSync (index.js:149)
    at FileReader.reader.onload (fileChooser.component.ts:44)
```

Reply:
I think we could support a range of compressions.

Reply:
Unzipping `.gz`-files is working in the web and standalone-version of codecharta.
It seems like webpack is providing some sort of polyfill for zlib. Running the e2e tests after building with `npm run build:web` works. `npm run build` is not working, because the application is build for a NodeEnvironment, which can't be served on the server provided from puppeteer.

I suggest we merge this branch.